src/process_utility.c

```diff
@@ -1426,6 +1425,35 @@ process_altertable_chunk(Hypertable *ht, Oid chunk_relid, void *arg)
 	AlterTableInternal(chunk_relid, list_make1(cmd), false);
 }
 
+static void
+process_altertable_set_tablespace_end(Hypertable *ht, AlterTableCmd *cmd)
+{
+	Oid			tspc_oid = get_rel_tablespace(ht->main_table_relid);
+	NameData	tspc_name;
+	Tablespaces *tspcs;
+
+	Assert(OidIsValid(tspc_oid));
+	namestrcpy(&tspc_name, cmd->name);
+
+	tspcs = tablespace_scan(ht->fd.id);
+	if (tspcs->num_tablespaces > 1)
+		ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("cannot attach new tablespace when multiple tablespaces are attached to hypertable \"%s\"",
```

Comment:
Sorry looking at the test output I think attach is confusing here. It should be set `cannot set new tablespace when multiple tablespaces are attached to hypertable \"%s\"`